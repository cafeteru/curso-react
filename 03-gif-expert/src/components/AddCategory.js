import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const defaultValue = "";
  const [inputValue, setInputValue] = useState(defaultValue);
  const handleInputChange = (e) => setInputValue(e.target.value);
  const handleSubmit = (event) => {
    event.preventDefault(); // Evita recarga de la pagina cuando envia el formulario
    if (inputValue.trim().length > 2) {
      setCategories((categories) => [inputValue, ...categories]);
      setInputValue(defaultValue);
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

AddCategory.propTypes = { setCategories: PropTypes.func.isRequired };
