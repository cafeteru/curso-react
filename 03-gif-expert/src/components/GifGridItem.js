export const GifGridItem = ({ title, url }) => {
  return (
    <div className={"card"}>
      <img alt={title} src={url} />
      <p>{title}</p>
    </div>
  );
};
