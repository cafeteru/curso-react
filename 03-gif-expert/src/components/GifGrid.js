import { useFetchGifs } from "../hooks/UseFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { gifs, loading } = useFetchGifs(category);
  return (
    <>
      <h3>{category}</h3>
      {loading && <p>"Cargando..."</p>}
      <div className={"card-grid"}>
        {gifs.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
