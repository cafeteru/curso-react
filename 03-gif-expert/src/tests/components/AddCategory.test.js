import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe("AddCategory", () => {
  const setCategories = jest.fn(); // crea un mock
  const value = "Hola mundo";
  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de cambiar la caja de texto", () => {
    const input = wrapper.find("input");
    input.simulate("change", { target: { value } });
  });

  test("No debe de postear la informaciÃ³n con submit", () => {
    wrapper.find("form").simulate("submit", {
      preventDefault() {},
    });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("Debe de llamar al setCategories y limpiar la caja de texto", () => {
    const input = wrapper.find("input");
    input.simulate("change", { target: { value } });
    wrapper.find("form").simulate("submit", {
      preventDefault() {},
    });
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(input.prop("value")).not.toBe(value);
    expect(input.prop("value")).toBe("");
  });
});
