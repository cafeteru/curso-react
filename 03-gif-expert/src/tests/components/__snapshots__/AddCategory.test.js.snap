// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory Debe de cambiar la caja de texto 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;

exports[`AddCategory Debe de llamar al setCategories y limpiar la caja de texto 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;

exports[`AddCategory No debe de postear la informaci√≥n con submit 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
